pkg/diagnostics/http_monitoring.go

```diff
@@ -229,3 +231,54 @@ func (h *httpMetrics) FastHTTPMiddleware(next fasthttp.RequestHandler) fasthttp.
 		h.ServerRequestCompleted(ctx, method, path, status, respSize, elapsed)
 	}
 }
+
+func (h *httpMetrics) convertPathToMethodName(path string) string {
+	var parsedPath = strings.Split(path, "/")
```

Comment:
Could the logic be possibly simple in parsing ?  If there is more than "/" in the string , skip else replace the last index of "\" from backwards to "*". That way we will avoid len checking and switch cases...

Reply:
Here are two cases: 

GET /v1/state/statestore/`key`
POST /v1/state/statestore

How can we know if `key` exists or not?

Reply:
my bad :-(  . we can not.